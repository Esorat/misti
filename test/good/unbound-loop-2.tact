fun test(): Int {
    let a: Int = 20; // read-only variable
    do  { // error: a is not changing within the loop
    } until (a > 10);
    return a;
}
