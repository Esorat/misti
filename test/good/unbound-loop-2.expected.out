test/good/unbound-loop-2.tact:4:14:
  3 |     do  { // error: a is not changing within the loop
> 4 |     } until (a > 10);
                   ^
  5 |     return a;
Unbounded Loop: The condition variable doesn't change within the loop
Help: Consider changing the variable within the loop to ensure it terminates
See: https://nowarp.github.io/tools/misti/docs/detectors/UnboundLoops